### DISCUSSION TITLE
Benefits of React.js?

### DISCUSSION THREAD

**Discussion by TansyEliot** [flybywiresim/aircraft#2983] [Q&A] [answered]
Hello, I saw that you are switching practically all systems to React.js.

1. Why? 
2. What are the benefits? 
3. Best performance/fps?

**Comment by Benjozork** [comment, marked as answer]
> Why?

The current system that MSFS uses for avionics is extremely 1. inefficient 2. hard to understand. It groups code together in an extremely unusual way that does not follow any best practice for JS development.

> What are the benefits?

* Better code readability
* Better code modularity
* Reduces the need to manually optimize re-rendering of unchanged elements

> Best performance/fps?

Yes, we expect performance to improve as a result of this - we are carefully considering a lot of aspects while rewriting the avionics.
